corehq/apps/locations/templates/locations/manage/location.html

```diff
@@ -83,6 +83,25 @@
       </script>
 
       {% if not form.is_new_location %}
+
+        <p class="lead">{{ location.name }} <small>{{ location.location_type_name }}</small></p>
```

Comment:
I could throw the location type name in a bootstrap badge or label, or maybe just `<small class="text-muted" />`